src/index.ts

```diff
@@ -129,58 +130,55 @@ class DefaultVisualizer {
             dataset.environments,
         );
 
-        this.map = new PropertiesMap(
-            config.map,
-            this._indexer,
-            dataset.properties,
-        );
-
-        this.map.active = this.structure.active;
-
         if (config.loadStructure !== undefined) {
             this.structure.loadStructure = config.loadStructure;
         }
 
+        this.structure.activeChanged = (guid, indexes) => {
+            this.map.setActive(guid);
```

Comment:
Is it kosher to set up these actions wrt the map before initializing said map?

Reply:
Since the callback will not be called before everything is initialized, I don't see this as a problem. I find the code easier to follow if everything related to the structure is close together

Reply:
I don't disagree, but it is a bit disconcerting to write operations on things that don't yet exist.